app/src/main/java/com/example/dinhh/soundscape/presentation/screens/record/RecordActivity.kt

```diff
@@ -0,0 +1,123 @@
+package com.example.dinhh.soundscape.presentation.screens.record
+
+import android.Manifest
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.view.View
+import com.example.dinhh.soundscape.R
+import android.content.pm.PackageManager
+import android.media.MediaPlayer
+import android.media.MediaRecorder
+import android.os.Build
+import android.support.v4.content.ContextCompat
+import android.util.Log
+import android.widget.Toast
+import kotlinx.android.synthetic.main.activity_record.*
+import java.io.File
+import java.io.IOException
+
+class RecordActivity : AppCompatActivity() {
+
+    private val RECORD_AUDIO_REQUEST_CODE = 101
+    private var fileName: String? = null
+    private var mPlayer: MediaPlayer? = null
+    private var myAudioRecorder: MediaRecorder? = null
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_record)
+        btnStopRecording.visibility = View.INVISIBLE
+        btnPlay.visibility = View.INVISIBLE
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            getPermissionToRecordAudio()
+        }
+
+        btnStartRecording.setOnClickListener {
+            btnStartRecording.visibility = View.INVISIBLE
+            btnStopRecording.visibility = View.VISIBLE
+            startRecording()
+        }
+
+        btnStopRecording.setOnClickListener {
+            btnStartRecording.visibility = View.VISIBLE
+            btnStopRecording.visibility = View.INVISIBLE
+            if (btnPlay.visibility == View.INVISIBLE){
```

Comment:
Write an extension for this, so you can use `btnPlay.isVisible()`instead of `btnPlay.visibility == View.INVISIBLE`

Check: https://github.com/soundscapeproject/soundscape/blob/develop/app/src/main/java/com/example/dinhh/soundscape/common/UIExt.kt

Reply:
Fixed